### PR TITLE
Make sure Bazel generated files get overwritten

### PR DESCRIPTION
**Pull request by pcmoritz**
The files generated by Bazel in `bazel-genfiles/` are not writeable, so after they are copied into `$ROOT_DIR/python` they will stay non-writeable and then the next time we try to copy them, they won't be copied. So we make them writeable here. This is a hack but better than not updating files.

The files need to be copied because Bazel doesn't allow us to produce build products outside of `bazel-genfiles/`, therefore it has to be with a followup copy command (with cmake we used an install directive to do this).

@ericl Any ideas to do this better?

cc @stephanie-wang

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -122,7 +122,7 @@ else
       --target=$ROOT_DIR/python/ray/pyarrow_files pyarrow==0.12.0.RAY \
       --find-links https://s3-us-west-2.amazonaws.com/arrow-wheels/9357dc130789ee42f8181d8724bee1d5d1509060/index.html
   bazel build //:ray_pkg -c opt
-  # Copy files and do not overwrite ownership
+  # Copy files and keep them writeable.
```
Comment (robertnishihara) [inline build.sh]: If it's a hack, can you add a note explaining that it's a hack and why we're doing it?

Comment (pcmoritz) [reply build.sh]: I added the comment.

**Comment by stephanie-wang** [comment]
Are all of the relevant files under `bazel-genfiles` symlinks? If so, maybe we can just copy the files to `python/`, but preserve the symlinks? Although I'm not sure how this would work if a local change causes a new file to be created (I don't think we would be able to copy this unless the new file's directory is made writable).

Also, by the way, should we be using `rsync` instead of `cp`?

**Comment by pcmoritz** [comment]
Concerning symlinks, I think we need to copy the files because if the user does "setup.py install" and delete the repo or bazel cache or wherever the actual files reside we need to make sure that the package still works. But we can investigate this more in the future.

I did consider rsync, but didn't want to introduce another dependency. The copying is sufficiently fast that this is ok I think (in CMake we also copied).

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**